core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -914,7 +914,7 @@ struct PatternImplSpecializationFromTag<Kokkos::ParallelForTag, Args...>
 
 template <class... Args>
 struct PatternImplSpecializationFromTag<Kokkos::ParallelReduceTag, Args...>
-    : identity<ParallelReduce<Args...>> {};
+    : identity<ParallelReduce<Args..., void>> {};
```

Comment:
Why did you change this?

Reply:
I didn't bother to try to figure out how to make the detection work for `Graph` kernels, here, and we just pass the original arguments here and force querying the functor for the value type. This is fixed in #5332.
I could look into fixing it here, too, if required/requested.